torch/_library/opaque_object.py

```diff
@@ -81,6 +96,12 @@ def get_payload(opaque_object: torch._C.ScriptObject) -> Any:
         payload (Any): The Python object stored in the opaque object. This can
         be set with `set_payload()`.
     """
+    if isinstance(opaque_object, FakeScriptObject):
+        raise ValueError(
+            "OpaqueObjects are opaque, so therefore the contents should not be "
+            "visible to torch.compile, and the fake kernel should not depend "
+            "on the contents of the OpaqueObject at all."
+        )
```

Comment:
nit:
- can deduplicate this with the below
- ValueError("get_payload: this function was called with a FakeScriptObject. The implication is that you are calling get_payload inside of a fake kernel. The fake kernel should not depend on the contents of the OpaqueObject at all, so we're erroring out. if you need this functionality, consider creating a custom TorchBind Object instead (but note that this is more difficult, <link to tutorial>)") (something like that that has more context)